### PR TITLE
Add IDisposable support to Mailosaur

### PR DESCRIPTION
**Pull request by dantheman999301**
Fairly simple PR just to add support for `IDisposable` for the `MailClient`.

### CODE REVIEW

File: Mailosaur.Test/ErrorsTests.cs [hunk 1]
```diff
@@ -9,36 +9,36 @@ public class ErrorsTests
         [Fact]
         public void UnauthorizedTest()
         {
-            var client = new MailosaurClient("invalid_key");
-            var ex = Assert.Throws<MailosaurException>(delegate
+            MailosaurException ex;
```
Comment (hippyhorse) [inline Mailosaur.Test/ErrorsTests.cs:12]: Thanks for this contribution! Please could you just change the syntax of each of these exception tests as follows, we can then merge:

```csharp
using (var client = new MailosaurClient("invalid_key"))
{
    var ex = Assert.Throws<MailosaurException>(delegate
    {
        client.Servers.List();
    });

    Assert.Equal("Authentication failed, check your API key.", ex.Message);
}
```

File: Mailosaur.Test/ErrorsTests.cs [hunk 2]
```diff
@@ -9,36 +9,36 @@ public class ErrorsTests
...
 
             Assert.Equal("Authentication failed, check your API key.", ex.Message);
         }
 
         [Fact]
         public void NotFoundTest()
         {
-            var client = new MailosaurClient(Environment.GetEnvironmentVariable("MAILOSAUR_API_KEY"));
-            var ex = Assert.Throws<MailosaurException>(delegate
+            MailosaurException ex;
```
Comment (hippyhorse) [inline Mailosaur.Test/ErrorsTests.cs:24]: See other comment

File: Mailosaur.Test/ErrorsTests.cs [hunk 3]
```diff
@@ -9,36 +9,36 @@ public class ErrorsTests
...
             Assert.Equal("Not found, check input parameters.", ex.Message);
         }
 
         [Fact]
         public void BadRequestTest()
         {
-            var client = new MailosaurClient(Environment.GetEnvironmentVariable("MAILOSAUR_API_KEY"));
-            var options = new ServerCreateOptions("");
-            var ex = Assert.Throws<MailosaurException>(delegate
+            MailosaurException ex;
```
Comment (hippyhorse) [inline Mailosaur.Test/ErrorsTests.cs:36]: See other comment

**Review by hippyhorse — CHANGES_REQUESTED** [review]
Minor change required to Exception tests

**Comment by hippyhorse** [comment]
Merged as part of https://github.com/mailosaur/mailosaur-dotnet/commit/463a10f00a7d96f016a73ad92ae87f16f028e4d5

**Comment by dantheman999301** [comment]
@hippyhorse apologies, I've only just seen this! Thanks for changing the tests

**Pull request closed without merging**